nixos/modules/services/networking/sabnzbd/default.nix

```diff
@@ -0,0 +1,609 @@
+                      description = ''
+                        Time, in seconds, to wait for a response before
+                        attempting error recovery.
+                      '';
+                      default = 60;
+                    };
+                    ssl = mkOption {
+                      type = types.bool;
+                      description = ''
+                        Whether the server supports TLS
+                      '';
+                      default = true;
+                    };
+                    ssl_verify = mkOption {
+                      type =
+                        types.coercedTo
+                          (types.enum [
+                            "strict"
+                            "allow injection"
+                            "none"
+                          ])
+                          (
+                            val:
+                            if val == "strict" then
+                              3
+                            else if val == "allow injection" then
+                              2
+                            else
+                              0
+                          )
+                          (
+                            types.enum [
+                              0
+                              1
+                              2
+                              3
+                            ]
+                          );
+
+                      description = ''
+                        Level of TLS verification. Supported values:
+                        3, 'strict'          -- strict (normal) verification
+                        2, 'allow injection' -- allow locally injected certificates
+                        1, 0, 'none'         -- no verification
+                      '';
+                      default = "strict";
+                    };
+                    expire_date = mkOption {
+                      type = types.nullOr types.str;
+                      description = ''
+                        If Notifications are enabled and an expiry date is
+                        set, warn 5 days before expiry. This setting
+                        does not automatically disable the server.
+                        Expected format: yyyy-mm-dd
+                      '';
+                      default = null;
+                    };
+                  };
+                }
+              );
+            };
+          };
+        };
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    warnings =
+      if cfg.configFile != null then
+        [
+          ''
+            `sabnzbd.configFile` is deprecated, consider using `sabnzbd.settings` instead.
+            If you have values set in `sabnzbd.settings` set, they will be ignored.
+          ''
+        ]
+      else
+        [ ];
+
+    users.users = mkIf (cfg.user == "sabnzbd") {
+      sabnzbd = {
+        uid = config.ids.uids.sabnzbd;
```

Comment:
Can we deprecate the hardcoded IDs?

Reply:
Will that cause any issues? I didn't write the original module so I'm not sure what the ramifications are

Reply:
Read up on nixos uid management a bit and it seems that existing users maintain their uids even if the config is changed, so that seems to be fine. Removed the hardcoded options in favor of `isSystemUser`.